tests/unit_tests/scheduler/test_scheduler.py

```diff
@@ -204,6 +205,44 @@ async def wait():
     assert timeouteventfound
 
 
+@pytest.mark.parametrize("max_running", [0, 1, 2, 10])
+async def test_max_running(max_running, mock_driver, storage, tmp_path):
+    runs: List[bool] = []
+
+    async def wait():
+        nonlocal runs
+        runs.append(True)
+        await asyncio.sleep(0.01)
```

Comment:
Can this be replaces with `await asyncio.sleep(0)`?

Reply:
I'd say await `asyncio.sleep(0.1)` just to be sure

Reply:
I put a small number there to let the async event loop have some time to let all realizations being started. But I tried with 0 and the test passed for at least 500 realizations on local iron. I think I will keep it at 0.01 as an insurance against flakiness. Bumping to 0.1 I will do if we see flakiness on this, but I think 0.01 is sufficient when even zero works fine.